sys/drv/bcm2835_emmc.c

```diff
@@ -113,21 +129,24 @@ static inline int bcmemmc_check_intr(bcmemmc_state_t *state) {
  * \return 0 on success, EIO on internal error, ETIMEDOUT on device timeout.
  * \warning This procedure may put the thread to sleep.
  */
-static int32_t bcmemmc_intr_wait(device_t *dev, uint32_t mask) {
+static emmc_error_t bcmemmc_intr_wait(device_t *dev, uint32_t mask) {
   bcmemmc_state_t *state = (bcmemmc_state_t *)dev->state;
 
   assert(mask != 0);
 
+  if (bcmemmc_invalid_state(state))
+    return bcmemmc_set_error(state, EMMC_ERROR_INVALID_STATE);
```

Comment:
(Feel free to ignore.) I find the description of the API presented in `emmc.h` a little confusing. It describes the return value as if only a single code could be returned at once (i.e. as if error codes could be encoded using binary numbers instead of one hot). I think it would be helpful to mention in the header that the return code is actually a mask of error codes.

Reply:
I've addede an extra note in emmc.h that explains it.